### ISSUE TITLE
Function with the same type already defined

### ISSUE THREAD

**Issue by a1880** [MiniZinc/libminizinc#323]
Using the MiniZinc IDE 2.3.2 under Windows, I am getting the following error for several examples:

```
MiniZinc: type error: 
  function with the same type already defined in E:/AK/tmp/MiniZinc IDE (bundled)-MiWdQh/untitled_model.mzn:2.1
```

The shrunk model:

```
function string: showBool(var bool: x) =
    if fix(x) then "1" else "0" endif;

var bool: a;

output [showBool(a) ++ showBool(a)];
```

The problem does not occur, if **showBool()** is only referenced once in the **output** statement.

**Comment by a1880** [comment]
This error is present again in release 2.4.0 102790580 (Windows 10)

**Comment by guidotack** [comment]
Actually it's a different bug this time (but it presents as the same error). I've pushed a fix, it will be available in the next bug fix release.

**Comment by a1880** [comment]
Actually fixed in 2.4.1

Thanks for the instant repair!
